### PR TITLE
Various updates throughout examples...

### PR DESCRIPTION
**Pull request by iamvery**
I worked through the whole set of koans this evening and made some tweaks that seem to improve the overall experience and learning.

### CODE REVIEW

**Comment by wolfbrain** [comment]
:eyes: @nathanl

File: about_atoms.exs [hunk 1]
```diff
@@ -32,14 +36,14 @@ defmodule AboutAtoms do
   end
 
   think "It is surprising to find out that booleans are atoms" do
-    assert_? Koans.Check.atom?(true)
-    assert_? Koans.Check.atom?(false)
+    assert_? is_atom(true)
```
Comment (nathanl) [inline about_atoms.exs:39]: Nice - I like that we now expose the user to the underlying function so they can learn it

File: about_matching.exs [hunk 2]
```diff
@@ -33,6 +33,20 @@ defmodule AboutMatching do
...
 
+  think "Matched values overwrite variable values" do
+    a = 1
+    assert a == __?
+
+    [a, _] = [2, 3]
+    assert a == __?
+  end
+
+  think "Values can be pinned to prevent them from being overwritten" do
```
Comment (nathanl) [inline about_matching.exs:44]: I think of pinning more as a way to assert what the righthand value looks like

**Comment by nathanl** [comment]
👍

File: about_matching.exs (hunk 2 above)
Comment (iamvery) [inline about_matching.exs:44]: I like that, so maybe "Values can be pinned to ensure the right-hand value meets your expectation"?

Comment (nathanl) [inline about_matching.exs:44]: Sounds good!

Comment (nathanl) [inline about_matching.exs:44]: It's kind of like when you do `{:ok, result} = some_func`; you want it to fail if the first item wasn't `:ok`.

Comment (iamvery) [inline about_matching.exs:44]: Right, it's like when you want to do that with some variable value.

Comment (iamvery) [inline about_matching.exs:44]: The more I think about this, I'm leaning back to how it's currently written. The "meets your expectation" is more _why_ you would pin a variable. The example describes the actual behavior. I wonder if there's a way to work both of those into the description...

Comment (iamvery) [inline about_matching.exs:44]: Could add a `# Note:` comment at the bottom of the example that says something about the expectation.

File: about_matching.exs [hunk 3]
```diff
@@ -33,6 +33,22 @@ defmodule AboutMatching do
...
+    a = 1
+    assert_raise MatchError, fn -> ^a = __? end
+    assert_raise MatchError, fn -> [^a, _] = __? end
+    # Note: This is a way of asserting what the right-hand side of the match
+    # meets your expectation. Similar to: {:ok, result} = some_func
```
Comment (iamvery) [inline about_matching.exs]: @nathanl here's that comment. What do you think?

**Pull request merged**